casanovo/config.py

```
@@ -26,6 +26,20 @@
 )
 
 
+def _int_or_float(value: Union[int, float]) -> Union[int, float]:
+    """Cast to an int (steps) or a float in [0, 1] (fraction of an epoch)."""
+    if isinstance(value, bool):
+        raise ValueError("must be an int or a float, not a bool")
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)